### PR TITLE
[cluster-agent] add k8s namespace cluster corecheck

### PR DESCRIPTION
**Pull request by fisherevans**
### What does this PR do?

Collects details on the k8s namespaces defined within the cluster

### Motivation

This is a often referenced resource type in K8S - but is missing from our current product dataset.

### Additional Notes

The namespace resource is very minimal - these changes were primarily a replication of a similarly bare-bones resource: role bindings.

It depends on https://github.com/DataDog/agent-payload/pull/177

### Describe how to test/QA your changes

To be tested on all supported versions of k8s

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by hestonhoffman — APPROVED** [review]

File: releasenotes/notes/collect-k8s-namespaces-332ee0d0625f4791.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - Collect K8S namespaces within the cluster agent
```
Comment (mfpierre) [inline releasenotes/notes/collect-k8s-namespaces-332ee0d0625f4791.yaml:10]: Actually this one should live in the `releasenotes-dca` folder

**Review by mfpierre — APPROVED** [review]
one nit but overall LGTM

File: releasenotes-dca/notes/collect-k8s-namespaces-332ee0d0625f4791.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - Collect K8S namespaces
```
Comment (xlucas) [inline releasenotes-dca/notes/collect-k8s-namespaces-332ee0d0625f4791.yaml:10]: We tend to categorize new resource collection as a new feature (also really minor but it's maybe clearer for the readers to not use short names in the changelog _Kubernetes_ vs _K8s_)

**Review by xlucas — APPROVED** [review]
LGTM

File: pkg/collector/corechecks/cluster/orchestrator/collectors/k8s/namespace.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	processor *processors.Processor
+}
+
+// NewNamespaceCollector creates a new collector for the Kubernetes
+// Namespace resource.
+func NewNamespaceCollector() *NamespaceCollector {
+	return &NamespaceCollector{
+		metadata: &collectors.CollectorMetadata{
+			IsDefaultVersion: true,
+			IsStable:         true,
```
Comment (xlucas) [inline pkg/collector/corechecks/cluster/orchestrator/collectors/k8s/namespace.go:44]: This should be false at start.

File: releasenotes-dca/notes/collect-k8s-namespaces-332ee0d0625f4791.yaml (hunk 2 above)
Comment (fisherevans) [reply releasenotes-dca/notes/collect-k8s-namespaces-332ee0d0625f4791.yaml:10]: this is good context, thanks

File: pkg/collector/corechecks/cluster/orchestrator/collectors/k8s/namespace.go (hunk 3 above)
Comment (fisherevans) [reply pkg/collector/corechecks/cluster/orchestrator/collectors/k8s/namespace.go:44]: Ah - that's the flag for enabled by default - thanks

**Pull request merged**